### PR TITLE
ships CSS-in-JS as part of each Web Component's dep graph

### PR DESCRIPTION
**Pull request by samthor**
⚠️ Not urgent. Does not fix a particular issue. But is "Part 3" of Rob and my plan to do codesplitting.

Moves component CSS to be included via PostCSS, which injects it into the page's `<head>` when the code is first loaded.

This isn't really my preferred solution as it adds the CSS before we know if a page is actually _using_ its related Web Component, but now I'm really worrying about the minutiæ. :) We can revisit this if we ever use Shadow DOM, which means we'd probably inline CSS or do something different.

This just moves code around. It shrinks our CSS bundle by ~40k but adds (as a string containing CSS) about ~20k to the measure page's JS, and ~20k to the default JS bundle. So the total benefit is you save about ~17% (since CSS was ~115k) if you only want one page.

This will have more benefits as (if?) we add more entrypoints.

### CODE REVIEW

**Comment by samthor** [comment]
This is a big change so I'll wait for @robdodson to take a look.

**Comment by samthor** [comment]
I mentioned this in the top comment but just to restate: this helps us in our move to Shadow DOM, in that it makes loading CSS a job of each component.

(Currently the plugin inserts the CSS dynamically, but it shouldn't be too hard to insert as part of the Shadow Root).

File: src/styles/components/_header.scss [hunk 1]
```diff
@@ -1,28 +1,16 @@
...
-@import '../../../styles/settings/colors';
-@import '../../../styles/settings/global';
-@import '../../../styles/tools/breakpoints';
-@import '../../../styles/tools/mixins';
-
-// Disable when Web Components aren't loaded; the button does nothing.
-body.unresolved web-header .web-header__hamburger-btn {
-  visibility: hidden;
-}
+@import '../settings/colors';
```
Comment (robdodson) [inline src/styles/components/_header.scss]: Do you think we should add back the license header to this and the other .scss files? My hunch is OSPO would like that.

Comment (samthor) [reply src/styles/components/_header.scss]: I agree but I can follow-up with a PR.

File: src/styles/all.scss [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
 @import 'components/all';
 @import 'pages/all';
-@import '../lib/components/all';
 @import 'overrides/all';
+
+// Elements marked with the unresolved class are hidden until the JS is first
+// loaded. Disable linter as it expects classes to only be ".w-...".
+// sass-lint:disable class-name-format
+*:not(body).unresolved {
+  display: none !important;
```
[the remark this answers is not in the document]
Comment (samthor) [reply src/styles/all.scss]: Ah yeah, I see—the hamburger especially as it appearing pushes everything around. I'll take a look.

[the remark this answers is not in the document]
Comment (samthor) [reply src/styles/all.scss]: I've tweaked the unresolved styles.

**Pull request merged**